api_shiftme/app/models/shift.rb

```diff
@@ -0,0 +1,2 @@
+class Shift < ApplicationRecord
```

Comment:
Did you forget something?I think that this is missing:
has_many : current_Shift_shift , class_name: 'Shift' , foreign_key: current_Shift_id
has_many : requested_Shift_shift , class_name: 'Shift' , foreign_key: requested_Shift_id

Reply:
I recommend use lowercase letter, putting "_" between words in react is enough